experimental/dds/tree2/src/test/shared-tree/editing.spec.ts

```diff
@@ -1012,6 +1038,199 @@ describe("Editing", () => {
 
 				expectJsonTree([tree, tree2], [{ foo: "x" }]);
 			});
+
+			it("can put a constraint on newly inserted node", () => {
+				const tree = makeTreeFromJson([{}]);
+				const tree2 = tree.fork();
+
+				const rootPath: UpPath = {
+					parent: undefined,
+					parentField: rootFieldKeySymbol,
+					parentIndex: 0,
+				};
+
+				// Insert foo
+				// State should be: [{ foo: {}}]
+				const sequence = tree.editor.sequenceField({
+					parent: rootPath,
+					field: brand("foo"),
+				});
+				sequence.insert(0, singleTextCursor({ type: jsonObject.name }));
+
+				// Constrain on foo existinga and insert into root sequence if it does
+				// State should be: [{ foo: {}}, {}]
+				tree.transaction.start();
+				tree.editor.addNodeExistsConstraint({
+					parent: rootPath,
+					parentField: brand("foo"),
+					parentIndex: 0,
+				});
+				const rootSequence = tree.editor.sequenceField({
+					parent: undefined,
+					field: rootFieldKeySymbol,
+				});
+				rootSequence.insert(1, singleTextCursor({ type: jsonObject.name }));
+				tree.transaction.commit();
+
+				// Make a concurrent edit to rebase over that inserts into root sequence
+				// State should be (to tree2): [{}, {}]
+				const tree2RootSequence = tree2.editor.sequenceField({
+					parent: undefined,
+					field: rootFieldKeySymbol,
+				});
+				tree2RootSequence.insert(1, singleTextCursor({ type: jsonObject.name }));
+
+				tree.merge(tree2);
+				tree2.rebaseOnto(tree);
+
+				expectJsonTree([tree, tree2], [{ foo: {} }, {}, {}]);
+			});
+
+			it.skip("constraint on newly inserted node with ancestor deleted concurrently", () => {
```

Comment:
Is this test meaningfully different from the other ancestor delete test? Was this supposed to be showing that a constraint that a node inserted in the same transaction exists can be violated if the insert fails due to an ancestor being deleted?

Reply:
That was the intent, but I think by swapping the order that makes the tests not meaningfully different if I'm understanding correctly.